cheetah/particles/parameter_beam.py

```diff
@@ -35,17 +35,16 @@ def __init__(
         factory_kwargs = {"device": device, "dtype": dtype}
         super().__init__()
 
-        self.register_buffer("_mu", torch.as_tensor(mu, **factory_kwargs))
-        self.register_buffer("_cov", torch.as_tensor(cov, **factory_kwargs))
-        total_charge = (
-            total_charge
-            if total_charge is not None
-            else torch.tensor([0.0], **factory_kwargs)
-        )
-        self.register_buffer(
-            "total_charge", torch.as_tensor(total_charge, **factory_kwargs)
-        )
-        self.register_buffer("energy", torch.as_tensor(energy, **factory_kwargs))
+        self.register_buffer("_mu", None)
+        self.register_buffer("_cov", None)
+        self.register_buffer("energy", None)
+        self.register_buffer("total_charge", torch.tensor(0.0, **factory_kwargs))
+
+        self._mu = torch.as_tensor(mu, **factory_kwargs)
+        self._cov = torch.as_tensor(cov, **factory_kwargs)
+        self.energy = torch.as_tensor(energy, **factory_kwargs)
+        if total_charge is not None:
+            self.total_charge = torch.as_tensor(total_charge, **factory_kwargs)
```

Comment:
When you register a buffer and pass a `torch.nn.Parameter` for initialization, the parameter is not properly registered and does not appear in `Module.parameters()` but `Module.buffers()`. Therefore I switched to a late initialization to make `torch.nn.Module` handle the parameter registration properly.
Passing `None` is necessary since the second argument is mandatory and does not have a default.

An alternative would be to check whether the passed arguments are `torch.Tensor` or `torch.nn.Paramter` and then call either `register_buffer()` or `register_parameter` appropriately.

Reply:
Ah, I think I got confused ... so
```python
self.register_buffer("_mu", torch.as_tensor(mu, **factory_kwargs))
self.register_buffer("_cov", torch.as_tensor(cov, **factory_kwargs))
self.register_buffer("energy", torch.as_tensor(energy, **factory_kwargs))
self.register_buffer("total_charge", torch.as_tensor(total_charge, **factory_kwargs))
```
would basically produce the original bug, right?

Reply:
Exactly. But I think we can actually do without manual registration for the arguments without default.

Reply:
Ok, no. We do need to manually register the buffers. Only `nn.Parameter` is registered automatically.